### PR TITLE
debug rpc: debug_setL1Head and better l1 timestamp management

### PR DESCRIPTION
**Pull request by tynes**
## Description

Adds a new RPC endpoint `debug_setL1Head` that can be used to reset the tip of the `SyncService`'s L1 syncing process.
Also adds in resetting the `LatestL1{BlockNumber,Timestamp]` when calling `debug_setHead`. These can be used to recover from failures more easily.

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](./CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: eth/api_backend.go [hunk 1]
```diff
@@ -104,8 +105,36 @@ func (b *EthAPIBackend) GetDiff(block *big.Int) (diffdb.Diff, error) {
...
+
+	tx := txs[0]
+	blockNumber := tx.L1BlockNumber()
+	if blockNumber == nil {
+		log.Error("No L1BlockNumber found in transaction", "number", number)
+		return
+	}
+
+	b.eth.syncService.SetLatestL1Timestamp(tx.L1Timestamp())
+	b.eth.syncService.SetLatestL1BlockNumber(blockNumber.Uint64())
```
Comment (karlfloersch) [inline eth/api_backend.go:133]: In future this could also call `SetL1Head(latestL1BlockNumber)`

File: les/api_backend.go [hunk 2]
```diff
@@ -91,6 +91,10 @@ func (b *LesApiBackend) SetHead(number uint64) {
 	b.eth.blockchain.SetHead(number)
 }
 
+func (b *LesApiBackend) SetL1Head(number uint64) {
+	panic("unimplemented")
```
Comment (karlfloersch) [inline les/api_backend.go:95]: Is there any way to trigger this panic at all that we haven't accounted for?

**Review by karlfloersch — APPROVED** [review]
Dope this LGTM!

File: les/api_backend.go (hunk 2 above)
Comment (tynes) [reply les/api_backend.go:95]: Nope, the light client server isn't used at all

**Pull request merged**